src/modules/Elsa.Alterations/Workflows/ExecuteAlterationPlanWorkflow.cs

```diff
@@ -32,8 +33,15 @@ protected override void Build(IWorkflowBuilder builder)
                     Result = new(planId)
                 },
                 new Correlate(planId),
-                new GenerateAlterationJobs(planId),
-                new DispatchAlterationJobs(planId),
+                new GenerateAlterationJobs(planId)
+                {
+                    Result = new(jobCount)
+                },
+                new If(context => jobCount.Get(context) > 0)
```

Comment:
Can we check if the correct jobs have been created instead of just relying on the count?

Reply:
Why?

Reply:
If we can guarantee that there will be no duplicates created than the count should be enough. However if we can not guarantee it we should check to see if the correct plans have been created.

Reply:
This logic is about determining if the Plan has any jobs or not. Because of no jobs were created, the plan should complete immediately. If at least one job was created, then we will wait for it to complete. Even if duplicate jobs are created (of which I have seen no evidence or suspect logic), we only care about the number of jobs in order to know if we need to complete the plan or wait for the jobs to complete.

Given this context, do you still see a potential problem with this logic?